### PR TITLE
Feature/1685 intersect x y coordinates of tracks with line sections

### PR DESCRIPTION
**Pull request by randy-seng**
OP#1685
OP#1860

### CODE REVIEW

**Review by briemla — CHANGES_REQUESTED** [review]
Looks good so far. One general comment: Validation of domain objects is done without an assert in the test. It might be useful to test attributes of the domain object. E.g. `test_geometry`.
The branch can be merged after questions have been cleared.

File: tests/OTAnalytics/application/test_eventlist.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        mock_event = Mock(spec=Event)
+
+        mock_intersector.intersect.return_value = mock_event
+
+        section_action_detector = SectionActionDetector(
+            mock_intersector, mock_section_event_builder
+        )
+        result_event = section_action_detector._detect_enter(line_section, track)
+        assert len(mock_section_event_builder.method_calls) == 3
+        assert len(mock_intersector.method_calls) == 1
```
[the remark this answers is not in the document]
Comment (randy-seng) [reply tests/OTAnalytics/application/test_eventlist.py:84]: Yes we should!😊

File: tests/OTAnalytics/application/test_eventlist.py [hunk 2]
```diff
@@ -80,6 +80,11 @@ def test_detect_enter(self, line_section: LineSection, track: Track) -> None:
             mock_intersector, mock_section_event_builder
         )
         result_event = section_action_detector._detect_enter(line_section, track)
+
+        mock_section_event_builder.add_section_id.assert_called()
+        mock_section_event_builder.add_event_type.assert_called()
+        mock_section_event_builder.add_direction_vector.assert_called()
+        mock_intersector.intersect.assert_called()
         assert len(mock_section_event_builder.method_calls) == 3
         assert len(mock_intersector.method_calls) == 1
```
Comment (briemla) [inline tests/OTAnalytics/application/test_eventlist.py]: Do we still need does two lines?

**Review by briemla — DISMISSED** [review]

File: tests/OTAnalytics/application/test_eventlist.py (hunk 2 above)
Comment (briemla) [reply tests/OTAnalytics/application/test_eventlist.py]: You can merge the PR afterwards

**Pull request merged**